framework/test/gtest/AnyMapTest.cpp

```diff
@@ -189,4 +195,21 @@ TEST(AnyMapTest, AnyMap)
   ASSERT_EQ(stream1.str(), "{do : 1, re : 2}");
   any_value_to_json(stream2, o_anymap1);
   ASSERT_EQ(stream2.str(), "{\"do\" : 1, \"re\" : 10}");
+}
+
+TEST(AnyMapTest, Move)
+{
+  AnyMap o(AnyMap::ORDERED_MAP);
+  o["do"] = Any(1);
+  o["re"] = Any(2);
+  AnyMap o_anymap_move_ctor(std::move(o));
+  ASSERT_EQ(any_cast<int>(o_anymap_move_ctor.at("do")), 1);
+  ASSERT_DEATH(o.size(), ".*")
```

Comment:
This behavior is not part of the C++ standard, so why do we need to implement it?

Reply:
I wanted a test to confirm that we are actually moving the guts of `o` to `o_anymap_move_ctor` - so any operation of `o` must crash because it's underlying pointer is a `nullptr`. The original problem this PR is trying to fix is a copy was performed even when we explicitly say `std::move` (because there was no move ctor). I wanted to verify we actually fix that.